src/main/java/uk/gov/hmcts/reform/jobscheduler/services/jobs/SchedulerExceptionHandlingHelper.java

```diff
@@ -0,0 +1,29 @@
+package uk.gov.hmcts.reform.jobscheduler.services.jobs;
+
+import org.quartz.SchedulerException;
+import uk.gov.hmcts.reform.jobscheduler.services.jobs.exceptions.JobException;
+
+public class SchedulerExceptionHandlingHelper {
```

Comment:
If it has only static methods and a private constructor, most likely it is a utility class that merely groups methods. In order to prevent instantiation, the class could've been marked as `final`.

Reply:
It can't be instantiated as it has a private constructor.
`final` would prevent extending this class btw...

Reply:
yup:
```
  <rule ref="rulesets/java/design.xml">
    <exclude name="UseUtilityClass"/>
```

Reply:
no harm in adding `final`

Reply:
Extending functionality can be done by modifying the class. No need to instantiate, no need to subclass, no need to override. It makes sense to have it marked as `final`.

Reply:
@KarolJastrzebski just clarity what `final` does as it's not what you described in your first comment, I would not extend this class :)